api/web_hub.go

```diff
@@ -115,84 +110,23 @@ func (h *Hub) Start() {
 					}
 				}
 
-			case channelId := <-h.invalidateChannel:
-				for webCon := range h.connections {
-					webCon.InvalidateCacheForChannel(channelId)
-				}
-
 			case msg := <-h.broadcast:
 				for webCon := range h.connections {
-					if shouldSendEvent(webCon, msg) {
-						select {
-						case webCon.Send <- msg:
-						default:
-							close(webCon.Send)
-							delete(h.connections, webCon)
-						}
+					if webCon.Broadcast(msg) {
+						delete(h.connections, webCon)
+						webCon.Close()
```

Comment:
I've removed this.  Don't think it was needed, and change it to a blocking call.